sherpa/instrument.py

```diff
@@ -717,10 +717,13 @@ def get_kernel(self, data, subkernel=True):
     def _check_pixel_size(self, data):
         if hasattr(self.kernel, "sky"):
             # This corresponds to the case when the kernel is actually a psf image, not just a model.
-            psf_pixel_size = self.kernel.sky.cdelt
-            data_pixel_size = data.sky.cdelt
-
-            if not numpy.allclose(psf_pixel_size, data_pixel_size):
-                warnings.warn("NOTE: The PSF pixel size ({}) does not correspond to the Image Pixel Size ({})".format(
-                    psf_pixel_size, data_pixel_size
-                ))
+            try:
+                psf_pixel_size = self.kernel.sky.cdelt
+                data_pixel_size = data.sky.cdelt
+
+                if not numpy.allclose(psf_pixel_size, data_pixel_size):
+                    warnings.warn("NOTE: The PSF pixel size ({}) does not correspond to the Image Pixel Size ({})"
+                                  .format(psf_pixel_size, data_pixel_size
+                    ))
+            except:
+                warnings.warn("Cannot determine the PSF or Data pixel size, ignoring.")
```

Comment:
If I were to see the warning:

```
Cannot determine the PSF or Data pixel size, ignoring.
```

I wouldn't know what was being ignored. I think this is meant to mean:

```
Unable to check whether the PSF and data pixel sizes match.
```

but maybe it's more (or less)?

Reply:
I used the text suggested by @cdeil